src/signaling.ts

```diff
@@ -29,15 +29,17 @@ class Signaling extends EventEmitter {
       ws.addListener("sdpNeeded", (event) => this.emit("sdpNeeded", event));
       ws.addListener("addIceCandidate", (event) => this.emit("addIceCandidate", event));
       ws.addListener("endpointRemoved", (event) => this.emit("endpointRemoved", event));
+      
+      const hasMediaPreferences = false;
```

Comment:
I'm far from a JS expert but wouldn't we want this to be a member variable like `ws` and `pingInterval`?

Reply:
Great question, I am also not a JS expert, so I'm doing some reading. @cejohnson-bw feel free to weigh in.

Reply:
I am also not a JS expert, but yes, I would move this up with the others. In practice, I think this would be ok since `connect` should only get called once, and reconnects should be handled under the covers by `ws`. However there is no question if it gets moved, so I would do that.

Reply:
Thanks!